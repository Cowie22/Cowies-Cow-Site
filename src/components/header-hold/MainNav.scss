.main-nav {
  padding: 24px 0;
  background: linear-gradient(180deg, rgba(0, 0, 0, 0.4) 0%, rgba(62, 64, 76, 0) 100%);
  z-index: 100;
  position: fixed;
  width: 100%;
  transition: all 1.2s;
  transition-delay: .2s;
  &.dark-nav {
    background: $dark-grey;
    transition: all 1.2s;
    transition-delay: .2s;
  }
  .header-logo-container {
    img {
      width: 65%;
      height: auto;
    }
  }
  .header-nav-container {
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;
    height: 100%;
    ul {
      position: relative;
      display: flex;
      justify-content: space-around;
      width: 100%;
      padding: 0 0 0px 0;
      margin: 8px 0 0 0;
      border-bottom: 1px solid #FFFFFF;
      list-style: none;
      a {
        color: $white;
        padding-bottom: 16px;
        width: 50%;
        height: 100%;
        font-weight: 600;
        font-size: 18px;
        line-height: 24px;
        text-align: center;
        letter-spacing: 0.01em;
        transition: color .5s;
        transition-delay: .2s;
        &:hover {
          cursor: pointer;
          color: $gold;
          transition: color .5s;
          transition-delay: .2s;
        }
        &:active {
          cursor: pointer;
          color: $gold;
          transition: color .5s;
          transition-delay: .2s;
        }
      }
      .active-link {
        position: absolute;
        bottom: -1px;
        left: 0;
        width: 50%;
        background: #E3FF39;
        height: 4px;
        cursor: pointer;
        transition: all .5s;
        transition-delay: .1s;
        &.home {
          left: 0;
          transition: all .5s;
          transition-delay: .1s;
        }
        &.about {
          left: 50%;
          transition: all .5s;
          transition-delay: .1s;
        }
        &.error {
          display: none;
        }
      }
      .active-link-hovered {
        position: absolute;
        bottom: -1px;
        width: 0;
        height: 0;
        cursor: pointer;
        transition: all .5s;
        transition-delay: .1s;
        &.home {
          left: 0;
          background: #E3FF39;
          height: 4px;
          width: 50%;
          transition: all .5s;
          transition-delay: .1s;
        }
        &.about {
          left: 50%;
          background: #E3FF39;
          height: 4px;
          width: 50%;
          transition: all .5s;
          transition-delay: .1s;
        }
      }
    }
  }
  // .secondary-nav {
  //   padding-top: 16px;
  //   display: flex;
  //   align-items: flex-start;
  //   justify-content: flex-start;
  //   &.none {
  //     display: none;
  //   }
  //   ul {
  //     display: flex;
  //     align-items: center;
  //     justify-content: space-around;
  //     list-style: none;
  //     padding: 0;
  //     margin: 0;
  //     li {
  //       a {
  //         width: fit-content;
  //         font-size: 14px;
  //         line-height: 16px;
  //         color: $white;
  //         text-align: center;
  //         border-bottom: 1px solid transparent;
  //         transition: all .5s;
  //         transition-delay: .2s;
  //         &:first-of-type {
  //           margin-right: 40px;
  //         }
  //       }
  //       &:hover {
  //         a {
  //           cursor: pointer;
  //           color: $gold;
  //           border-bottom: 1px solid $gold;
  //           transition: all .5s;
  //           transition-delay: .2s;
  //         }
  //       }
  //     }
  //   }
  // }
}

@media only screen and (max-width: 992px) {
  .main-nav {
    padding: 16px 0;
    .header-logo-container {
      img {
        width: 100%;
        height: auto;
      }
    }
  }
}

// @media only screen and (max-width: 992px) {
//   .header-section-mobile {
//     padding: 16px 0;
//     background: linear-gradient(180deg, rgba(0, 0, 0, 0.4) 0%, rgba(62, 64, 76, 0) 100%);
//     z-index: 100;
//     position: fixed;
//     width: 100%;
//     transition: all 1.2s;
//     transition-delay: .2s;
//     &.dark-nav {
//       background: linear-gradient(180deg, #17181C 0%, #3E404C 39.06%);
//       transition: all 1.2s;
//       transition-delay: .2s;
//     }
//     .header-logo-container {
//       img {
//         width: 86%;
//         height: auto;
//       }
//     }
//     .mobile-navbar-container {
//       display: flex;
//       justify-content: flex-end;
//       align-items: center;
//       height: 100%;
//       .navbar-toggle {
//         display: block;
//         position: relative;
//         margin: 0;
//         height: 50px;
//         width: 43px;
//         overflow: hidden;
//         align-self: center;
//         background-color: transparent;
//         border: 0;

//         @media screen and (max-width: 400px) and (min-width: 320px) {
//           height: 43px;
//         }
    
//         &:active {
//           .icon-bar {
//             background: $gold;
//             .icon-bar-inner {
//               background: $gold;
//             }
//           }
//           &:after {
//             color: $gold;
//             .icon-bar-inner {
//               background: $gold;
//             }
//           }
//           &.active {
//             &:after {
//               color: $gold;
//               .icon-bar-inner {
//                 background: $gold;
//               }
//             }
//           }
//         }
    
//         &.active {
//           width: 100%;
//           .icon-bar {
//             width: 100%;
//           }
//           .icon-bar:nth-child(2) {
//             transform: rotate(45deg);
//             top: 18%;
//             left: 12%;
//           }
    
//           .icon-bar:nth-child(3) {
//             transform: translateX(-50px);
//           }
    
//           .icon-bar:nth-child(4) {
//             transform: rotate(-45deg);
//             top: 72%;
//             left: 12%;
//           }
//         }
    
//         &:focus {
//           outline: none;
//         }
    
//         .icon-bar {
//           margin: 0;
//           display: block;
//           background: $white;
//           width: 100%;
//           height: 1px;
//           border-radius: 2px;
//           position: absolute;
//           right: 0;
//           -webkit-transform: rotate(0deg);
//           -moz-transform: rotate(0deg);
//           -o-transform: rotate(0deg);
//           transform: rotate(0deg);
//           -webkit-transition: transform 300ms ease;
//           -moz-transition: transform 300ms ease;
//           -ms-transition: transform 300ms ease;
//           -o-transition: transform 300ms ease;
//           transition: transform 300ms ease;
//           -webkit-transform-origin: 1px;
//           -moz-transform-origin: 1px;
//           -o-transform-origin: 1px;
//           transform-origin: 1px;
//           .icon-bar-inner {
//             margin: 0;
//             display: block;
//             background: $white;
//             width: 50%;
//             height: 4px;
//             border-radius: 2px;
//             position: absolute;
//             left: 25%;
//             top: -1px;
//             -webkit-transform: rotate(0deg);
//             -moz-transform: rotate(0deg);
//             -o-transform: rotate(0deg);
//             transform: rotate(0deg);
//             -webkit-transition: transform 300ms ease;
//             -moz-transition: transform 300ms ease;
//             -ms-transition: transform 300ms ease;
//             -o-transition: transform 300ms ease;
//             transition: transform 300ms ease;
//             -webkit-transform-origin: 1px;
//             -moz-transform-origin: 1px;
//             -o-transform-origin: 1px;
//             transform-origin: 1px;
//           }
//         }
    
//         .icon-bar:nth-child(2) {
//           top: 16%;
//         }
    
//         .icon-bar:nth-child(3) {
//           top: 44%;
//         }
    
//         .icon-bar:nth-child(4) {
//           top: 72%;
//         }
//       }
//     }
//     .header-nav-container-mobile {
//       margin: 24px 0 40px 0;
//       .header-nav-list {
//         border-left: 1px solid $white;
//         list-style: none;
//         padding: 0;
//         margin: 0;
//         height: auto;
//         transform: scaleY(1);
//         transition: transform .7s;
//         transition-delay: .1s;
//         transform-origin: top;
//         .home-page-list-items {
//           border: none;
//           padding: 0 16px 12px 16px;
//           &.active-list {
//             border-left: 4px solid $gold;
//           }
//         }
//         .about-page-list-items {
//           border: none;
//           padding: 12px 16px 0 16px;
//           &.active-list {
//             border-left: 4px solid $gold;
//           }
//         }
//         li {
//           width: 100%;
//           text-align: center;
//           font-weight: 600;
//           font-size: 14px;
//           line-height: 16px;
//           letter-spacing: 0.01em;
//           padding: 24px 0 0 0;
//           a {
//             color: $white;
//           }
//           &:active {
//             a {
//               cursor: pointer;
//               color: $gold;
//               border-bottom: 1px solid $gold;
//               transition: all .5s;
//               transition-delay: .2s;
//             }
//           }
//           &.mobile-page-nav-item {
//             background: rgba(255, 255, 255, 0.1);
//             font-size: 18px;
//             line-height: 24px;
//             padding: 8px 0;
//           }
//         }
//         .mobile-nav-lower-list {
//           border-left: 4px solid $gold;
//         }
//       }
//       &.hidden-dropdown {
//         margin: 0;
//         .header-nav-list {
//           height: 0;
//           transform: scaleY(0);
//           transition: transform 0.4s;
//         }
//       }
//     }
//   }
//   .scroll-none {
//     overflow: hidden;
//   }
// }
