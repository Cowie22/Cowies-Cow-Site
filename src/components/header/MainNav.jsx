import React, { useState, useEffect } from 'react'
import { Link } from 'gatsby'
import { Container, Row, Col } from 'react-bootstrap'
import AppProvider, { AppContext } from '../../contexts/state'

import ExternalLink from '../exitramp/ExternalLink.jsx'
import HeaderDropdown from '../HeaderDropdown/HeaderDropdown'
import HeaderDropdownData from '../HeaderDropdown/HeaderDropdownData'

import logo from '../../assets/images/header-logo.svg'
import upArrow from '../../assets/images/upArrow.svg'
import downArrow from '../../assets/images/downArrow.svg'


class MainNav extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentData: {list: ['placeholder']},
      isNavOpen: false,
      mobileDropDown: 0,
    }
  }

  toggleNav = () => {
    document.querySelector('.mobile-nav').scrollTop = 0;
    this.setState({
      isNavOpen: !this.state.isNavOpen,
    })
    let isiTray = document.getElementsByClassName('isi-tray')[0];
    if (this.state.isNavOpen) {
      document.body.classList.remove('scroll-none');
      isiTray.classList.remove('hide');
    } else {
      document.body.classList.add('scroll-none');
      isiTray.classList.add('hide');
    }
  }

  handleNavHover = () => {
    this.headerDropdownPromise()
      .then(currentData => this.headerDropdownPromise())
      .then(currentData => {
        this.handleDropdown(currentData, currentData.name)
      })
      .catch(err => console.log('There was an error:' + err))
  }

  headerDropdownPromise = () => {
    const { activeHeaderDropdown } = this.props;
    let updatedData = {list: ['placeholder']};

    return new Promise((resolve, reject) => {
      let dataFilter = HeaderDropdownData.filter((data, i) => {
        if (data.name === activeHeaderDropdown) {
          updatedData = data;
        }
      })
      dataFilter ? resolve(updatedData) : reject('ERROR');
    })
  }

  handleDropdown = (val, name) => {
    let dropdownContainer = document.getElementById('header-dropdown-container');
    dropdownContainer.classList = `header-dropdown-container ${name}`;
    this.setState({
      currentData: val,
    })
  }

  handleMobileDropDown = (val) => {
    this.setState({
      mobileDropDown: val,
    })
  }

  render() {
    const { currentData, isNavOpen, mobileDropDown } = this.state;
    const {
      activeHeaderDropdown,
      handleActiveHeaderDropdown,
      activeDropdownLink,
      handleActiveDropdownLink,
      currentPage,
      handleCurrentPage,
    } = this.props;
    return (
      <>
        <section className='main-nav'
          onMouseLeave={() => {
            handleActiveHeaderDropdown('');
            this.handleNavHover()
          }}
        >
          <Container>
            <Row>
              <Col lg={3} xs={12} className='navbar-header'>
                <Link
                  to='/'
                >
                  <img
                    loading='lazy'
                    alt='Myovant logo'
                    className='header-logo'
                    src={logo}
                    onMouseEnter={() => {
                      handleActiveHeaderDropdown('');
                      this.handleNavHover();
                    }}
                  />
                </Link>
                <button
                  type='button'
                  className={isNavOpen ? 'navbar-toggle active' : 'navbar-toggle'}
                  onClick={() => {
                    this.toggleNav()
                  }}
                >
                  <span className='sr-only'>Toggle navigation</span>
                  <span className='icon-bar' />
                  <span className='icon-bar' />
                  <span className='icon-bar' />
                </button>
              </Col>
              <Col xl={{span: 3, offset: 3}} lg={{span: 3, offset: 2}} className='d-none d-lg-block'>
                <nav className='header-nav-container'>
                  <ul>
                    <li
                      onMouseEnter={() => {
                        handleActiveHeaderDropdown('resources');
                        this.handleNavHover();
                      }}
                      onClick={() => {
                        handleActiveHeaderDropdown('');
                        handleActiveDropdownLink(0);
                      }}
                    >
                      <Link to='/resources/'>
                        Access & Resources
                      </Link>
                      <img src={activeHeaderDropdown === 'resources' ? upArrow : downArrow} className='nav-arrow' />
                    </li>
                    {/* <li
                      onMouseEnter={() => {
                        handleActiveHeaderDropdown('test');
                        this.handleNavHover();
                      }}
                    >
                      TEST
                    </li> */}
                  </ul>
                </nav>
              </Col>
              <Col
                xl={{span: 3, offset: 0}}
                lg={{span: 4, offset: 0}}
                xs={{span: 6, offset: 1}}
                className='d-none d-lg-block'
              >
                <div
                  className='header-call-now-container'
                  onMouseEnter={() => {
                    handleActiveHeaderDropdown('');
                    this.handleNavHover();
                  }}
                >
                  <div className='cta-btn-container header-cta-btn-container'>
                    <a>
                      <button className='cta-btn pink-btn'>
                        Register for Launch Broadcast
                      </button>
                    </a>
                  </div>
                </div>
              </Col>
            </Row>
              <div
                className={activeHeaderDropdown === '' ? 'header-dropdown-hidden d-none d-lg-block' : 'header-dropdown d-none d-lg-block'}
                onMouseLeave={() => {
                  handleActiveHeaderDropdown('');
                  this.handleNavHover();
                }}
              >
                <HeaderDropdown
                  currentData={currentData}
                  activeDropdownLink={activeDropdownLink}
                  handleActiveDropdownLink={handleActiveDropdownLink}
                  handleActiveHeaderDropdown={handleActiveHeaderDropdown}
                  currentPage={currentPage}
                  handleCurrentPage={handleCurrentPage}
                />
              </div>
          </Container>
        </section>
        <section className={isNavOpen ? 'mobile-nav opened' : 'mobile-nav'}>
          <Container>
            <Row>
              <Col className='mobile-nav-col'>
                <div
                  className='mobile-nav-list-container'
                  className={
                    mobileDropDown !== 0 ? 'mobile-nav-list-container' : 'mobile-nav-list-container hidden'
                  }
                >
                  <ul className='mobile-nav-links'>
                    <li
                      // onClick={() => {
                      //   this.toggleNav()
                      // }}
                    >
                      <div
                        className='mobile-nav-dropdown-link'
                        onClick={() => {
                          mobileDropDown !== 1 ? this.handleMobileDropDown(1) : this.handleMobileDropDown(0)
                        }}
                      >
                        Access & Resources
                        <img src={mobileDropDown === 1 ? upArrow : downArrow} className='nav-arrow' />
                      </div>
                      <ul
                        className={
                          mobileDropDown === 1 ? 'header-dropdown-ul' : 'header-dropdown-ul hidden'
                        }
                      >
                        <li
                          onClick={() => {
                            handleActiveDropdownLink(0);
                            handleActiveHeaderDropdown('');
                            this.handleMobileDropDown(0)
                            this.toggleNav();
                          }}
                          className={(activeDropdownLink === 0 && currentPage === 'resources') ? 'header-dropdown-active' : ''}
                        >
                          <Link to='/resources/'>
                            Access & Resources
                          </Link>
                        </li>
                        <li
                          onClick={() => {
                            handleActiveDropdownLink(1);
                            handleActiveHeaderDropdown('');
                            this.handleMobileDropDown(0)
                            this.toggleNav();
                          }}
                          className={(activeDropdownLink === 1 && currentPage === 'resources') ? 'header-dropdown-active' : ''}
                        >
                          <Link to='/resources/#resource-access-content'>
                            Access Information
                          </Link>
                        </li>
                        <li
                          onClick={() => {
                            handleActiveDropdownLink(2);
                            handleActiveHeaderDropdown('');
                            this.handleMobileDropDown(0)
                            this.toggleNav();
                          }}
                          className={(activeDropdownLink === 2 && currentPage === 'resources') ? 'header-dropdown-active' : ''}
                        >
                          <Link to='/resources/#resource-patient-content'>
                            Patient Support Program
                          </Link>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </div>
              </Col>
            </Row>
          </Container>
          <div className='mobile-nav-overlay d-block d-lg-none'>&nbsp;</div>
          </section>
          <section className='mobile-register-btn d-block d-lg-none'>
            <a href=''>
              <Container>
                <Row>
                  <Col>
                    <p className='white text-center'>
                      Register for Launch Broadcast
                    </p>
                  </Col>
                </Row>
              </Container>
            </a>
          </section>
      </>
    )
  }
}

export default MainNav