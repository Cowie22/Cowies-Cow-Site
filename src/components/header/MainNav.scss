.main-nav {
  padding: 16px 0;
  background: $white;
  .header-logo {
    height: 94px;
  }
  .header-call-now-container {
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    .header-cta-btn-container {
      width: 100%;
    }
  }
  .nav-container {
    justify-content: flex-start;
    ul {
      position: relative;
      display: flex;
      justify-content: center;
      align-items: center;
      @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
        margin-left: -14px;
      }
      li {
        margin-right: 0px;
        text-align: center;
        transition: 0.2s color;
        width: 100%;
        font-family: $mulish;
        &:hover {
          color: $red;
          transition: 0.2s color;
        }
        .header-discussion-container {
          display: flex;
          &:hover {
            font-weight: normal;
          }
          .header-discussion-icon {
            height: 51px;
            margin-right: 8px;
            @media all and (-ms-high-contrast: none),
              (-ms-high-contrast: active) {
              height: 50px;
              width: 60px;
            }
          }
        }
      }
      li:not(:last-of-type) {
        margin-right: 34px;
      }
      @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
        li:not(:last-of-type) {
          margin-right: 38px;
        }
        .ie-margin {
          margin-left: -30px;
          margin-right: 30px;
        }
      }
      @media only screen and (max-width: 1200px) and (min-width: 769px) {
        li:not(:last-of-type) {
          margin-right: 22px;
        }
        li {
          a {
            font-size: 14px;
          }
          .header-discussion-container {
            .header-discussion-icon {
              height: 44px;
            }
          }
        }
      }
    }
  }
  .active-header {
    width: 100%;
    height: 4px;
    background: $red;
    transition: 1s all;
    position: absolute;
    top: 70px;
  }
}

.mobile-nav {
  display: none;
}

.navbar-toggle {
  display: none;
}

@media only screen and (max-width: 992px) {
  .main-nav {
    padding: 8px 0;
    background-color: #ffffff;
  }

  .mobile-nav {
    margin-top: 11px;
    max-height: 100vh;
    &.opened {
      display: block;
      background-color: $white;
      width: 100%;
      transition: 0.2s ease;
    }
  }

  .mobile-nav-overlay {
    background-color: $white;
    background-size: cover;
    height: 100vh;
    z-index: 9999;
    position: relative;
  }

  .mobile-nav-links {
    display: flex;
    align-items: center;
    flex-direction: column;
    width: 100%;
    list-style: none;
    padding: 0;

    li {
      width: 100%;
      padding: 0 15px;
      transition: 0.2s all;

      &:hover {
        color: $red;
        transition: 0.2s color;
      }

      &:first-child {
        margin-top: 15px;
      }

      a {
        color: $black;
        display: block;
        padding: 10px 0;
        text-decoration: none;
        font-size: 16px;
        line-height: 18px;
      }
    }
  }

  .mobile-utility-links {
    border-top: 1px solid #ffffff;
    padding: 8px 0 0 0;
    list-style: none;

    li {
      padding: 0 15px;
      font-weight: 600;
      a {
        color: #ffffff;
        display: block;
        padding: 8px 0;
        font-size: 14px;
        transition: 0.2s all;
        &:hover {
          text-decoration: none;
          color: $red;
          transition: 0.2s color;
        }
      }
    }
  }

  @media only screen and (max-width: 768px) {
    .mobile-nav-col {
      padding-left: 0px;
      .header-discussion-container {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        border-top: 2px solid #eaeaea;
        border-bottom: 2px solid #eaeaea;
        padding: 24px 0;
        .header-discussion-icon {
          padding-left: 8px;
        }
      }
      .header-providers-container {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        .header-providers-icon {
          padding-left: 8px;
        }
      }
    }
  }

  .navbar-header {
    display: flex;
    justify-content: space-between;
    // .header-logo {
    //   width: 100%;
    // }
  }

  .navbar-toggle {
    display: block;
    position: relative;
    margin: 0;
    height: 43px;
    width: 42px;
    overflow: hidden;
    align-self: center;
    background-color: transparent;
    border: 0;

    &:active {
      .icon-bar {
        background: $red;
      }
      &:after {
        color: $red;
      }
      &.active {
        &:after {
          color: $red;
        }
      }
    }

    &.active {
      width: 45px;
      .icon-bar {
        width: 35px;
      }
      .icon-bar:nth-child(2) {
        transform: rotate(45deg);
        top: 0px;
      }

      .icon-bar:nth-child(3) {
        transform: translateX(-50px);
      }

      .icon-bar:nth-child(4) {
        transform: rotate(-45deg);
        top: 23px;
      }

      &:after {
        content: "CLOSE";
        position: absolute;
        right: 2px;
        bottom: 0;
        margin-top: 3px;
        color: $red;
        font-size: 12px;
        font-weight: 700;
        line-height: normal;
      }
    }

    &:after {
      content: "MENU";
      position: absolute;
      right: 2px;
      bottom: 0;
      margin-top: 3px;
      color: $red;
      font-size: 12px;
      font-weight: 700;
      line-height: normal;
    }

    &:focus {
      outline: none;
    }

    .icon-bar {
      margin: 0;
      display: block;
      background: $red;
      width: 42px;
      height: 2px;
      border-radius: 2px;
      position: absolute;
      right: 0;
      -webkit-transform: rotate(0deg);
      -moz-transform: rotate(0deg);
      -o-transform: rotate(0deg);
      transform: rotate(0deg);
      -webkit-transition: transform 300ms ease;
      -moz-transition: transform 300ms ease;
      -ms-transition: transform 300ms ease;
      -o-transition: transform 300ms ease;
      transition: transform 300ms ease;
      -webkit-transform-origin: 1px;
      -moz-transform-origin: 1px;
      -o-transform-origin: 1px;
      transform-origin: 1px;
    }

    .icon-bar:nth-child(2) {
      top: 1px;
    }

    .icon-bar:nth-child(3) {
      top: 10px;
    }

    .icon-bar:nth-child(4) {
      top: 19px;
    }
  }
}
