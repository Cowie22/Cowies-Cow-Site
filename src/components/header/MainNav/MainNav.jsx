import React, { useState, useEffect, useContext, memo } from 'react'
import { Link } from 'gatsby'
import { Container, Row, Col } from 'react-bootstrap'
import { AppContext } from '../../../contexts/state'

import useHover from '../../../customHooks/useHover'

import logo from '../../../assets/images/global/header-logo.svg'

const MainNav = (props) => {
  const [isHovered, hover] = useHover('')
  const state = useContext(AppContext)
  const { currentPage, updateCurrentPage } = state

  const [currentTabHovered, handleCurrentTabHovered] = useState('')

  return (
    <section className='main-nav'>
      <section className='navigation-container'>
        <Container>
          <Row>
            <Col lg={{ span: 3, offset: 0 }}>
              <div className='header-logo-container'>
                <Link to='/'>
                  <img loading='lazy' alt='' src={logo} />
                </Link>
              </div>
            </Col>
            <Col xl={{ span: 6, offset: 3 }} lg={{ span: 7, offset: 2 }}>
              <nav className='nav-container'>
                <div className='nav-item-container'>
                  <Link
                    href='/'
                    className={currentPage === 'home' ? 'active-page' : ''}
                  >
                    Home
                  </Link>
                  <Link
                    href='/patientsandcaregivers/'
                    className={currentPage === 'patients-and-caregivers' ? 'active-page' : ''}
                  >
                    Patients and caregivers
                  </Link>
                  <Link
                    href='/hcps/'
                    className={currentPage === 'healthcare-providers' ? 'active-page' : ''}
                  >
                    Healthcare providers
                  </Link>
                </div>
              </nav>
            </Col>
          </Row>
        </Container>
      </section>
    </section>
  )
}

export default memo(MainNav)
