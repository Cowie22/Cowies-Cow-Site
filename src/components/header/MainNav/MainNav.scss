.main-nav {
  background: $dark-grey;
  z-index: 100;
  position: fixed;
  top: 0;
  width: 100%;
  .navigation-container {
    padding: 40px 0 38px 0;
    .header-logo-container {
      height: 100%;
      display: flex;
      justify-content: flex-start;
      align-items: center;
      &:hover {
        cursor: pointer;
      } 
    }
    .nav-container {
      display: flex;
      justify-content: flex-end;
      align-items: center;
      height: 100%;
      .nav-item-container {
        display: flex;
        justify-content: space-between;
        a {
          color: $black;
          padding-right: 32px;
          transition: 0.5s color;
          font-weight: 400;
          font-size: 21px;
          line-height: 27px;
          &:last-of-type {
            padding-right: 0px;
          }
          &::after {
            display: block;
            content: '';
            border-bottom: 5px solid $gold;  
            transform: scaleX(0);  
            transition: transform 0.5s ease-in-out;
            transform-origin: 100% 50%;
          }
          &:hover {
            cursor: pointer;
            color: $gold;
            transition: 0.5s color;
            &::after {
              transform: scaleX(1);
              transform-origin: 0% 50%;
            }
          }
          &.active-page {
            &::after {
              display: block;
              content: '';
              border-bottom: 5px solid $gold;  
              transform: scaleX(1);
              transform-origin: 0% 50%;
              transition: transform 0.5s ease-in-out;
            }
          }
        }
      }
    }
  }
}