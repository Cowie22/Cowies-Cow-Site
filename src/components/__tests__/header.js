import React from 'react'
import { render } from '@testing-library/react'
import MainNav from '../header/MainNav'

const { debug, getByAltText, getByText, getByTestId, getAllByText, container } = render( <MainNav /> )

const myfembreeLogo = getByAltText('MyfembreeÂ® (relugolix, estradiol, and norethindrone acetate) tablets 40 mg, 1 mg, 0.5 mg')
const onceDailyDosing = getByTestId('Once-Daily Dosing')
const onceDailyDosingMobile = getByTestId('Once-Daily Dosing Mobile')
const efficacy = getByTestId('Efficacy')
const safety = getByTestId('Safety')
const safetyMobile = getByTestId('Safety Mobile')
const howItWorks = getByTestId('How It Works')
const howItWorksMobile = getByTestId('How It Works Mobile')
const support = getByTestId('Support & Savings')

test('Header links exist', () => {
  // debug()
  expect(myfembreeLogo).toBeTruthy()
  expect(onceDailyDosing).toBeTruthy()
  expect(onceDailyDosingMobile).toBeTruthy()
  expect(efficacy).toBeTruthy()
  expect(safety).toBeTruthy()
  expect(safetyMobile).toBeTruthy()
  expect(howItWorks).toBeTruthy()
  expect(howItWorksMobile).toBeTruthy()
  expect(support).toBeTruthy()
})

test('Header links have appropriate href', () => {
  expect(myfembreeLogo.parentElement.getAttribute('href')).toBe('/')
  expect(onceDailyDosing.getAttribute('href')).toBe('/once-daily-dosing/')
  expect(onceDailyDosingMobile.getAttribute('href')).toBe('/once-daily-dosing/')
  expect(efficacy.getAttribute('href')).toBe('/liberty-trials/')
  expect(safety.getAttribute('href')).toBe('/safety-profile/')
  expect(safetyMobile.getAttribute('href')).toBe('/safety-profile/')
  expect(howItWorks.getAttribute('href')).toBe('/how-it-works/')
  expect(howItWorksMobile.getAttribute('href')).toBe('/how-it-works/')
  expect(support.getAttribute('href')).toBe('/myfembree-support-program/')
})

// test('Header snapshot matches', () => {
//   return
//   // once the component is done, use snapshot
//   // expect(container.firstChild).toMatchSnapshot()
// })