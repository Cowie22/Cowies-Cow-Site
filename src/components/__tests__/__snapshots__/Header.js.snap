// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot matches 1`] = `null`;
